src/main/java/org/jabref/logic/pdf/search/indexing/IndexingTaskManager.java

```diff
@@ -72,6 +77,20 @@ private void enqueueTask(Runnable indexingTask) {
         }
     }
 
+    private void updateIndexName() {
+        Optional<Path> file = bibDatabaseContext.getDatabasePath();
+
+        StringBuilder indexTitle = new StringBuilder();
```

Comment:
A string builder is a bit of overkill here (it's used if you have to concatenate a lot of strings).
Something like the following should work
```suggestion
        String name = bibDatabaseContext.getDatabasePath()
                                        .map(file -> file.getFileName().toString())
                                        .orElse("untitled")
```